src/components/percentage/percentage.test.jsx

```diff
@@ -0,0 +1,261 @@
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Percentage {...requiredProps} id="unique-id" ref={ref} />);
+      expect(ref.current.id).toBe('unique-id');
+    });
+  });
+
+  describe('label API', () => {
+    it('can be set', () => {
+      render(<Percentage {...requiredProps} label="Unique label" />);
+      expect(screen.getByLabelText('Unique label')).toBeInTheDocument();
+    });
+  });
+
+  describe('name API', () => {
+    it('sets the name attribute', () => {
+      render(<Percentage {...requiredProps} name="test-name" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('name', 'test-name');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Percentage {...requiredProps} name="unique-name" ref={ref} />);
+      expect(ref.current.name).toBe('unique-name');
+    });
+  });
+
+  describe('onBlur API', () => {
+    it('sets the onblur handler', () => {
+      const onBlurSpy = jest.fn();
+      render(<Percentage {...requiredProps} onBlur={onBlurSpy} />);
+      fireEvent.blur(screen.getByLabelText('the label'));
+      expect(onBlurSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('onChange API', () => {
+    it('sets the onchange handler', () => {
+      const onChangeSpy = jest.fn();
+      render(<Percentage {...requiredProps} onChange={onChangeSpy} />);
+      fireEvent.change(screen.getByLabelText('the label'), { target: { value: '100' } });
+      expect(onChangeSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('onFocus API', () => {
+    it('sets the onFocus handler', () => {
+      const onFocusSpy = jest.fn();
+      render(<Percentage {...requiredProps} onFocus={onFocusSpy} />);
+      fireEvent.focus(screen.getByLabelText('the label'));
+      expect(onFocusSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('onInput API', () => {
+    it('sets the onInput handler', () => {
+      const onInputSpy = jest.fn();
+      render(<Percentage {...requiredProps} onInput={onInputSpy} />);
+      fireEvent.input(screen.getByLabelText('the label'));
+      expect(onInputSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('onKeyDown API', () => {
+    it('sets the onKeyDown handler', () => {
+      const onKeyDownSpy = jest.fn();
+      render(<Percentage {...requiredProps} onKeyDown={onKeyDownSpy} />);
+      fireEvent.keyDown(screen.getByLabelText('the label'));
+      expect(onKeyDownSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('placeholder API', () => {
+    it('sets the placeholder attribute', () => {
+      render(<Percentage {...requiredProps} placeholder="test-placeholder" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('placeholder', 'test-placeholder');
+    });
+  });
+
+  describe('readOnly API', () => {
+    it('sets the readOnly attribute', () => {
```

Comment:
Since `readOnly` is a boolean, we should be exhaustive and also set it to `false` in a test.